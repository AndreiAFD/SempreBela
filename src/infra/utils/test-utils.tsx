import React from 'react';
import { render } from '@testing-library/react-native';
import {
  AppThemeContext,
  AppThemeContextProvider,
} from '@infra/contexts/themeContext';
import { PropsWithChildren } from 'react';
import { NavigationContainer } from '@react-navigation/native';

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return (
    <AppThemeContextProvider>
      <NavigationContainer>{children}</NavigationContainer>
    </AppThemeContextProvider>
  );
};

// <TranslationProvider messages={defaultStrings}>
//   {children}
// </TranslationProvider>

const customRender = (ui: React.ReactElement, options: any) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
