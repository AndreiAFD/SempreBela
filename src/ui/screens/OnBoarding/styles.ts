import {Platform, Touchable, TouchableOpacity} from 'react-native';
import {RFValue} from 'react-native-responsive-fontsize';
import styled from 'styled-components/native';
import * as Animatable from 'react-native-animatable';

export const Container = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
  padding-top: ${Platform.OS === 'android' ? RFValue(70) : 0}px;
`;

export const NameWrapper = styled(Animatable.View)`
  width: 100%;
  margin-top: ${Platform.OS === 'ios' ? RFValue(30) : RFValue(30)}px;
  padding: ${RFValue(24)}px;
  gap: 20px;
`;

export const Button = styled.TouchableOpacity`
  height: ${RFValue(45)}px;
  border-radius: ${RFValue(10)}px;
  background-color: white;
  margin: ${RFValue(33)}px ${RFValue(24)}px;
  align-items: center;
  justify-content: center;
`;
