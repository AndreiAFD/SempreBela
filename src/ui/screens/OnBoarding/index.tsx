import React from 'react';
import { StatusBar } from 'react-native';
import AnimatedLottieView from 'lottie-react-native';
import { Flex } from '@react-native-material/core';
import LinearGradient from 'react-native-linear-gradient';
import * as Animatable from 'react-native-animatable';
import { RFValue } from 'react-native-responsive-fontsize';
import { useTheme } from 'styled-components';
import Text from '@components/Text';
import WomanAnimation from '@assets/animations/woman-animation.json';
import * as S from './styles';

const OnBoarding = () => {
  const theme = useTheme();
  return (
    <LinearGradient
      colors={['#38B48A', '#01726B']}
      style={{ flex: 1 }}
      testID="onBoarding-container">
      <StatusBar backgroundColor="#38B48A" />
      <S.Container>
        <Animatable.View
          animation="fadeInDown"
          duration={1400}
          style={{ width: '100%', height: RFValue(270) }}>
          <AnimatedLottieView
            source={WomanAnimation}
            autoPlay
            loop
            style={{
              maxHeight: 270,
              marginTop: RFValue(20),
            }}
          />
        </Animatable.View>

        <S.NameWrapper animation={'fadeInLeft'} delay={700}>
          <Text fontWeight="bold" fontSize={RFValue(43)} color={'#ffffff'}>
            Salão Sempre
          </Text>
          <Text fontWeight="bold" fontSize={RFValue(43)} color={'#ffffff'}>
            Bela
          </Text>
          <Animatable.View animation="fadeInLeft" delay={800}>
            <Text
              fontWeight="regular"
              fontSize={RFValue(14)}
              color={'#ffffff'}
              marginTop={RFValue(16)}>
              There are many variations of passages of Lorem Ipsum available,
              but.
            </Text>
          </Animatable.View>
        </S.NameWrapper>
      </S.Container>

      <Animatable.View animation="fadeInUp" delay={1200}>
        <S.Button>
          <Text fontWeight="regular" color="#00726b">
            Cadastre-se
          </Text>
        </S.Button>
      </Animatable.View>
      <Animatable.View animation={'fadeInUp'} delay={1400}>
        <Flex direction="row" justify="center" mb={RFValue(33)}>
          <Text fontWeight="regular" color="white">
            {' '}
            Já tem uma conta ?
          </Text>
          <Text fontWeight="bold" color="white">
            {' '}
            Login
          </Text>
        </Flex>
      </Animatable.View>
    </LinearGradient>
  );
};

export default OnBoarding;
