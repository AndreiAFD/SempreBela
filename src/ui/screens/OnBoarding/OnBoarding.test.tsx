import React from 'react';
import { render, screen } from '@infra/utils/test-utils';
import OnBoarding from '@screens/OnBoarding';

jest.mock('react-native-linear-gradient');

describe('On Boarding screen', () => {
  it('should render correctly', () => {
    console.log(OnBoarding);
    render(<OnBoarding />, null);

    // const container = screen.getByTestId('onBoarding-container');

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
