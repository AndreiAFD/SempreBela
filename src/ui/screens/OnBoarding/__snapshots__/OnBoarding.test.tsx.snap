// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On Boarding screen should render correctly 1`] = `null`;
